fastify.js

```diff
@@ -721,9 +722,12 @@ function fastify (options) {
       const reply = new Reply(res, request, childLogger)
 
       if (disableRequestLogging === false) {
-        childLogger.info({ req: request }, 'incoming request')
+        if (createRequestLogMessage) {
+          childLogger.info(createRequestLogMessage(req))
```

Comment:
the user here can set only the 1st parameter of `childLogger.info`.

I would think another way to handle it like:

```
createRequestLogMessage(childLogger, req)
```

so it is up to the user choose to log info or warning (i used to change the log level based on the request headers)

Reply:
Ok perfect!  I will make sure to modify the parameters so that createRequestLogMessage will accept childLogger as a parameter.

Reply:
>so it is up to the user choose to log info or warning (i used to change the log level based on the request headers)

Hmm, that's a good one. But then I'd like to know if the initial signature is the best for the use case you describe.
The current issue the PR attempts to solve is to customize the log message only, as per #4463. If that's the goal, I think that it is a matter of just applying the correct order to the log statement. i.e.

```suggestion
          childLogger.info({ req: request }, createRequestLogMessage(req))
```

If something more complex is required, I believe the current documentation for [`disableRequestLogging`](https://fastify.dev/docs/latest/Reference/Server#disablerequestlogging) already covers well what t do in those cases. wdyt? 🙂